Codeinwp/optimole-wp
# Add Dashboard pages

### All Submissions:

 

### Changes proposed in this Pull Request:
- Adds Optimole dashboard page and subpages;
- The following pages are added: 
  - Main page -> leads to the dashboard;
  - Settings -> leads to the dashboard and autofocuses the **Settings** tab;
  - Cloud Library -> when cloud browsing feature is enabled;
- Adds mechanism to autofocus dashboard tabs when coming from an URL with the respective hash ( `#settings`, `#dashboard`, `#help`);
- The URL also changes when changing the tabs manually;
- When already in the dashboard, the links in the sidebar should navigate to the proper tab instead of reloading the page;
- Redirects old dashboard URL `upload.php?page=optimole` to new dashboard link: `admin.php?page=optimole`- this is done to make sure that old links in other plugins do not break.

![image](https://github.com/Codeinwp/optimole-wp/assets/15010186/0cb92444-053b-48f1-a074-207ffde13fad)


Closes Codeinwp/optimole-service#986.

### How to test the changes in this Pull Request:
1. Check that the links are added to the sidebar;
2. The redirect works fine: `upload.php?page=optimole` to the new dashboard page;
3. Check that navigating to the dashboard pages via sidebar links works ok both when the dashboard is loaded and when it's not;
4. Check that loading the page with the hashes loads the proper tab;
5. Make sure that when enabling and disabling the Cloud Browsing feature, the link is properly added to the sidebar;

### Other information:

 
**Note**: This is based on the `feat/dam` branch, hence the PR to that instead of `next`.
